boa_engine/src/context/intrinsics.rs

```diff
@@ -933,3 +955,302 @@ impl IntrinsicObjects {
+        );
+
+        let mut unmapped_arguments = ordinary_object.clone();
+
+        // // 4. Perform DefinePropertyOrThrow(obj, "length", PropertyDescriptor { [[Value]]: 𝔽(len),
+        // // [[Writable]]: true, [[Enumerable]]: false, [[Configurable]]: true }).
+        unmapped_arguments.property(
+            length_property_key,
+            Attribute::WRITABLE | Attribute::NON_ENUMERABLE | Attribute::CONFIGURABLE,
+        );
+
+        // // 7. Perform ! DefinePropertyOrThrow(obj, @@iterator, PropertyDescriptor {
+        // // [[Value]]: %Array.prototype.values%, [[Writable]]: true, [[Enumerable]]: false,
+        // // [[Configurable]]: true }).
+        unmapped_arguments.property(
+            JsSymbol::iterator().into(),
+            Attribute::WRITABLE | Attribute::NON_ENUMERABLE | Attribute::CONFIGURABLE,
+        );
+
+        let mut mapped_arguments = unmapped_arguments.clone();
+
+        // // 8. Perform ! DefinePropertyOrThrow(obj, "callee", PropertyDescriptor {
+        // // [[Get]]: %ThrowTypeError%, [[Set]]: %ThrowTypeError%, [[Enumerable]]: false,
+        // // [[Configurable]]: false }).
+        unmapped_arguments.accessor(
+            "callee".into(),
+            true,
+            true,
+            Attribute::NON_ENUMERABLE | Attribute::PERMANENT,
+        );
+
+        // 21. Perform ! DefinePropertyOrThrow(obj, "callee", PropertyDescriptor {
+        // [[Value]]: func, [[Writable]]: true, [[Enumerable]]: false, [[Configurable]]: true }).
+        mapped_arguments.property(
+            "callee".into(),
+            Attribute::WRITABLE | Attribute::NON_ENUMERABLE | Attribute::CONFIGURABLE,
+        );
+
+        let mut iterator_result = ordinary_object.clone();
+        iterator_result.property(
+            "value".into(),
+            Attribute::WRITABLE | Attribute::CONFIGURABLE | Attribute::ENUMERABLE,
+        );
+        iterator_result.property(
+            "done".into(),
+            Attribute::WRITABLE | Attribute::CONFIGURABLE | Attribute::ENUMERABLE,
+        );
+
+        Self {
+            iterator_result,
+            ordinary_object,
+            array,
+            number,
+            string,
+            symbol,
+            bigint,
+            boolean,
+            unmapped_arguments,
+            mapped_arguments,
+            function_with_prototype,
+            function_prototype,
+            function,
+            async_function,
+            function_without_proto,
+            function_with_prototype_without_proto,
+        }
+    }
+
+    /// Cached iterator result template.
+    ///
+    /// Transitions:
+    ///
+    /// 1. `__proto__`: `Object.prototype`
+    /// 2. `"done"`: (`WRITABLE`, `CONFIGURABLE`, `ENUMERABLE`)
+    /// 3. `"value"`: (`WRITABLE`, `CONFIGURABLE`, `ENUMERABLE`)
+    pub(crate) const fn iterator_result(&self) -> &ObjectTemplate {
+        &self.iterator_result
+    }
+    /// Cached ordinary object template.
```

Comment:
I would add line breaks between functions, to keep it cleaner. This is just one, but there are more below.
```suggestion
    }
    
    /// Cached ordinary object template.
```